cmd/frontend/graphqlbackend/org.go

```diff
@@ -283,6 +283,31 @@ func (r *schemaResolver) UpdateOrganization(ctx context.Context, args *struct {
 	return &OrgResolver{db: r.db, org: updatedOrg}, nil
 }
 
+// RemoveOrganization removes an org and all resources associated to it. Depending on the flag value, it will hard delete or soft delete the org.
+func (r *schemaResolver) RemoveOrganization(ctx context.Context, args *struct {
+	Organization graphql.ID
+}) (*EmptyResponse, error) {
+	if !envvar.SourcegraphDotComMode() {
+		return nil, errors.New("hard deleting organization is not supported")
+	}
+
+	orgDeletionFlag, err := r.db.FeatureFlags().GetFeatureFlag(ctx, "org-deletion")
+	if err != nil || orgDeletionFlag == nil || !orgDeletionFlag.Bool.Value {
+		return nil, errors.New("hard deleting organization is not supported")
+	}
```

Comment:
We are currently swallowing the underlying error:

```suggestion
	if err != nil {
		return nil, err
	} else if orgDeletionFlag == nil || !orgDeletionFlag.Bool.Value {
		return nil, errors.New("hard deleting organization is not supported")
	}
```